design-documents/elasticsearch/extended-analyzer-config.md

```diff
@@ -0,0 +1,278 @@
+    <toke_filters>
+        <default>
+            <lowercase /><!-- declare usage of standard filter -->
+            <my_token_filter><!-- customized token filter -->
+                <type>standard</type>
+                <max_token_length xsi:type="number">5</max_token_length>
+            </my_token_filter>
+        </defualt>
+        <en_US><!-- locale specific filters>
+            <!-- ... -->
+        </en_US>
+    </toke_filters>
+</config>
+```
+
+### Char Filters Configuration
+Char filters configuration is similar to token filters but declared by `char_filters` element
+
+```xml
+<config>
+    <char_filters>
+        <default>
+            <html_strip /><!-- declare usage of standard filter -->
+            <my_char_filter><!-- customized filter to replece "-" by "_" -->
+                <type>pattern_replace</type>
+                <pattern>(\\d+)-(?=\\d)</pattern>
+                <replacement>$1_</replacement>
+            </my_char_filter>
+        </defualt>
+        <en_US><!-- locale specific filters>
+            <!-- ... -->
+        </en_US>
+    </char_filters>
+</config>
+```
+
+### ElasticSearch parameters XML representation
+
+JSON configuration described at ElasticSearch configuration may be converted to XML configuration to specify tokenizer and filter parameters.
+
+#### Objects
+
+Objects or maps are key structure to declare ElasticSearch configuration. When converting JSON to XML keys became element names and values are presented as node values. An element may declare a type of value with `xsi:type` attribute but it may be autodetected by the parser.
+
+| JSON        | XML           | 
+| ------------- |-------------|
+| `{key: {}}`  | `<key xsi:type="map"><!-- ... --></key>` |
+
+
+#### Strings
+
+String values are converted to text node:
+
+| JSON        | XML           | 
+| ------------- |-------------|
+| `{key: "string value"}`  | `<key xsi:type="string">string value</key>` |
+
+#### Numbers
+
+Numeric values are converted to text node. This type utilize both integer and float numbers depend on provided decimal part:
+
+| JSON        | XML           | 
+| ------------- |-------------|
+| `{key: 42}`  | `<key xsi:type="number">42</key>` |
+
+#### Booleans
+
+Boolean values are converted to text node and may contain "true" or "false":
+
+| JSON        | XML           | 
+| ------------- |-------------|
+| `{key: true}`  | `<key xsi:type="boolean">true</key>` |
+
+#### Least
+
+Array values represented as series of `item` nodes:
+
+| JSON        | XML           | 
+| ------------- |-------------|
+| `{key: ['v1', 'v2']}`  | `<key xsi:type="list"><item>v1</item><item>v2</item></key>` |
+
+Item node may declare `ref` attribute that should be unique inside list. `ref` attribute should be used to provide a possibility to override list element by Magento configuration merging mechanism. If module B want to add element to list in ElasticSearch config declared by module A then `ref` attribute should be used as well.
```

Comment:
Should `ref` be `name` or it's different from how we use `name` in other configuration files?

Reply:
It may renamed to `name` for consistency.